trio/_path.py

```diff
@@ -182,6 +182,73 @@ async def open(self, *args, **kwargs):
         value = await trio.to_thread.run_sync(func)
         return trio.wrap_file(value)
 
+    if TYPE_CHECKING:
+        # the dunders listed in _forward_magic that aren't seen otherwise
+        __bytes__ = pathlib.Path.__bytes__
+        __truediv__ = pathlib.Path.__truediv__
+        __rtruediv__ = pathlib.Path.__rtruediv__
+
+        # Note: these are not parsed properly by jedi.
+        # It might be superior to just manually implement all the methods and get rid
+        # of all the magic wrapping stuff.
+
+        # wrapped methods handled by __getattr__
+        absolute: Any
+        as_posix: Any
+        as_uri: Any
+        chmod: Any
+        cwd: Any
+        exists: Any
+        expanduser: Any
+        glob: Any
+        home: Any
+        is_absolute: Any
+        is_block_device: Any
+        is_char_device: Any
+        is_dir: Any
+        is_fifo: Any
+        is_file: Any
+        is_reserved: Any
+        is_socket: Any
+        is_symlink: Any
+        iterdir: Any
+        joinpath: Any
+        lchmod: Any
+        lstat: Any
+        match: Any
+        mkdir: Any
+        read_bytes: Any
+        read_text: Any
+        relative_to: Any
+        rename: Any
+        replace: Any
+        resolve: Any
+        rglob: Any
+        rmdir: Any
+        samefile: Any
+        stat: Any
+        symlink_to: Any
+        touch: Any
+        unlink: Any
+        with_name: Any
+        with_suffix: Any
+        write_bytes: Any
+        write_text: Any
+
+        if sys.platform != "win32":
+            group: Any
+            is_mount: Any
+            owner: Any
+
+        if sys.version_info >= (3, 8):
+            link_to: Any
+        if sys.version_info >= (3, 9):
+            is_relative_to: Any
+            with_stem: Any
+            readlink: Any
+        if sys.version_info >= (3, 10):
+            hardlink_to: Any
+
```

Comment:
This line isn't the one I wanted to comment on, but I can't comment on it cause it's out of the diff github shows :(

Namely, the line that is `os.PathLike.register(Path)`. Presumably we could *subclass* `os.PathLike` and that would help some typing stuff? I don't know though.

Reply:
Definitely, `register` isn't going to be fully supported everywhere. It'd also be better to inherit to indicate that we provide string paths, not bytes. A little tricky though since it's only as of 3.9 that you could subscript `PathLike`:
```py
if TYPE_CHECKING:
    _StrPathLike = os.PathLike[str]
else:
    _StrPathLike = os.PathLike
...
class Path(_StrPathLike):
```

Reply:
let's leave that to a different PR that fully types `trio.Path`, can either continue with that in #2630 or open a different issue